src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title,price) VALUES(?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                         query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException("Creating book failed, no rows affected.");
+            }
+            book.setId(getCreatedId(statement));
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        Book book = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+
+            ResultSet generatedKeys = statement.executeQuery();
+            if (generatedKeys.next()) {
+                book = mapToBook(generatedKeys);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = mapToBook(resultSet);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a connection to the DB");
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        Long id = book.getId();
+        Optional<Book> optionalBook = findById(id);
+        if (optionalBook.isEmpty()) {
+            throw new RuntimeException("Book not found with id " + id);
+        }
```

Comment:
This is unnecessary work, we don't need to look for the book earlier. Our methods shouldn`t depend on each other